tools/log-collector/helper.go

```diff
@@ -153,62 +148,37 @@ func filterCSV(csvObjects unstructured.UnstructuredList) unstructured.Unstructur
 // filterCRD returns list of crds created by given set of groups
 func filterCRD(crdObjs unstructured.UnstructuredList) (unstructured.UnstructuredList, error) {
 	crdFilterGroup := []string{TriliovaultGroup, SnapshotStorageGroup, CsiStorageGroup}
-	var filteredCRDObject unstructured.UnstructuredList
+	var filteredCRDObjects unstructured.UnstructuredList
 	for index := range crdObjs.Items {
 		for in := range crdFilterGroup {
 			crdGroup, _, err := unstructured.NestedString(crdObjs.Items[index].Object, "spec", "group")
 			if err != nil {
 				log.Errorf("Unable to get the CRD Group field : %s", err.Error())
-				return filteredCRDObject, err
+				return filteredCRDObjects, err
 			}
 			if crdFilterGroup[in] == crdGroup {
-				filteredCRDObject.Items = append(filteredCRDObject.Items, crdObjs.Items[index])
-			}
-		}
-	}
-	return filteredCRDObject, nil
-}
-
-// getGVByGroup returns group_version matched for given group
-func getGVByGroup(apiGVList []*apiv1.APIGroup, groupName string, isPreferredVersion bool) (gvList []string) {
-
-	for index := range apiGVList {
-		if apiGVList[index].Name == groupName {
-			if isPreferredVersion {
-				gvList = append(gvList, apiGVList[index].PreferredVersion.GroupVersion)
-			}
-			for in := range apiGVList[index].Versions {
-				gvList = append(gvList, apiGVList[index].Versions[in].GroupVersion)
+				filteredCRDObjects.Items = append(filteredCRDObjects.Items, crdObjs.Items[index])
 			}
 		}
 	}
-	return gvList
+	return filteredCRDObjects, nil
 }
 
-// getResourcesGVByName resource object and gv for given resource name
-func getResourcesGVByName(resourceMap map[string][]apiv1.APIResource, name string) map[string]apiv1.APIResource {
+// filterNS returns list of Namespaces Object given by user input in --namespaces flag
+func filterNS(nsObjs unstructured.UnstructuredList, namespaces []string) unstructured.UnstructuredList {
+	var filteredNSObjects unstructured.UnstructuredList
 
-	gvResourceMap := make(map[string]apiv1.APIResource)
-	for gv, resource := range resourceMap {
-		for index := range resource {
-			if resource[index].Name == name {
-				gvResourceMap[gv] = resource[index]
-				continue
-			}
-		}
+	nsNames := make(sets.String)
+	for _, name := range namespaces {
```

Comment:
no need of this for loop... use `nsNames := sets.NewString(namespaces...)`

Reply:
resolved